lightning/src/ln/channelmanager.rs

```diff
@@ -7638,27 +7638,34 @@ where
 		if peer_state_mutex_opt.is_none() { return }
 		peer_state_lock = peer_state_mutex_opt.unwrap().lock().unwrap();
 		let peer_state = &mut *peer_state_lock;
+
+		let remaining_in_flight =
+			if let Some(pending) = peer_state.in_flight_monitor_updates.get_mut(funding_txo) {
+				pending.retain(|upd| upd.update_id > highest_applied_update_id);
+				pending.len()
+			} else { 0 };
+
 		let channel =
 			if let Some(ChannelPhase::Funded(chan)) = peer_state.channel_by_id.get_mut(channel_id) {
 				chan
 			} else {
+				if remaining_in_flight != 0 {
+					return;
+				}
```

Comment:
Can this be pulled out and done unconditionally prior to the `channel` assignment? We're checking it again later, which would be unnecessary if done earlier? IIUC, we'd only skip the logging below.